.github/scripts/simple-smoke-test.sh

```diff
@@ -1,49 +1,88 @@
 #!/usr/bin/env bash
 
-set -e
-set -o pipefail
+set -o errexit -o nounset -o pipefail
 
 function test_docker_image() {
     local image=$1
     local container_name=$2
     local expected_distribution_type=$3
+    local expected_version=$4
 
-    if [ "$(docker ps --all --quiet --filter name="$container_name")" ]; then
-      echo "Removing existing '$container_name' container"
-      docker container rm --force "$container_name"
+    remove_container_if_exists "${container_name}"
```

Comment:
pls add comment as to why we need to remove
This will run in its own docker/agent machine so I would assume env is fresh. I am sure I am missing something so comment will be good

Reply:
> pls add comment as to why we need to remove
This will run in its own docker/agent machine so I would assume env is fresh. I am sure I am missing something so comment will be good

This is existing functionality. I'm guessing it's from local debugging.